base/compiler/abstractinterpretation.jl

```diff
@@ -1482,11 +1482,10 @@ function abstract_invoke(interp::AbstractInterpreter, (; fargs, argtypes)::ArgIn
     types = rewrap_unionall(Tuple{ft, unwrap_unionall(types).parameters...}, types)::Type
     nargtype = Tuple{ft, nargtype.parameters...}
     argtype = Tuple{ft, argtype.parameters...}
-    result = findsup(types, method_table(interp))
-    result === nothing && return CallMeta(Any, false)
-    match, valid_worlds = result
-    method = match.method
+    match, valid_worlds = findsup(types, method_table(interp))
     update_valid_age!(sv, valid_worlds)
```

Comment:
If you return `Any`, there is no reason to need to update the valid worlds, since you did not use any information from the look up.

Reply:
I wondered if this allows us to invalidate unsuccessful inference trial when a new method comes to be available, e.g.
```julia
ulia> foo(::Int) = :Int
foo (generic function with 1 method)

julia> CC.findsup(Tuple{typeof(foo),Integer}, CC.method_table(CC.NativeInterpreter()))
(nothing, Core.Compiler.WorldRange(0x0000000000000000, 0xffffffffffffffff))

julia> callfoo(n) = Base.@invoke foo(n::Integer)
callfoo (generic function with 1 method)

julia> callfoo2(n) = callfoo(n)
callfoo2 (generic function with 1 method)

julia> @code_typed callfoo2(42)
CodeInfo(
1 ─ %1 = Core.invoke(Main.foo, Tuple{Integer}, n)::Any
└──      return %1
) => Any

julia> foo(::Integer) = :Integer
foo (generic function with 2 methods)

julia> CC.findsup(Tuple{typeof(foo),Integer}, CC.method_table(CC.NativeInterpreter()))
(Core.MethodMatch(Tuple{typeof(foo), Integer}, svec(), foo(::Integer) in Main at REPL[6]:1, true), Core.Compiler.WorldRange(0x0000000000007e7e, 0xffffffffffffffff))

julia> @code_typed callfoo2(42)
CodeInfo(
1 ─ %1 = Core.invoke(Main.foo, Tuple{Integer}, n)::Any
└──      return %1
) => Any
```

But as indicated at these lines, it seems like we don't pick up a valid minimum world age when the match failed, so this `update_valid_age!` would be really ineffective in this kind of case.
```julia
julia> CC.findsup(Tuple{typeof(foo),Integer}, CC.method_table(CC.NativeInterpreter()))
(nothing, Core.Compiler.WorldRange(0x0000000000000000, 0xffffffffffffffff))
```